#216 Allow ContentType queries to be performed on non-default databases.

rp-austin opened the pull request.

vdboor commented:
Thanks for the wonderful work, this seems like a ready to merge feature!

I wonder, instead or passing `using=DEFAULT_DB_ALIAS` to all model methods, would it work to read `sefl._state.db` or is the `ContentType` located in a different DB then the models that derive from it?

rp-austin commented:
Thanks for the feedback! I took your suggestion and switched to using `self._state.db` in the model methods. It keeps the interface cleaner and I think is more consistent with Django's behavior, so it seems like a better way to handle this.

(I didn't actually know about `ModelState` before your comment, which is why I wasn't using it before :) )

The one exception is the `save` method, and by extension the `pre_save_polymorphic` method. `save` checks the `using` kwarg, then `self._state.db`, then falls back on `DEFAULT_DB_ALIAS`; `pre_save_polymorphic` simply uses the kwarg value. Since `self._state.db` won't be populated for new instances and callers may also override the database when calling `save`, I thought it was better to use the argument. However, I can certainly change this behavior if you'd prefer.

I also fixed a couple spots in the `query_translate` functions where I wasn't passing the `using` value correctly before. `instance_of` filters should work correctly now.

Finally, I've also added a couple unit tests for the new functionality, but not many. If I need to add more, please let me know.

Thanks!

vdboor commented:
Wonderful, thanks for updating this swiftly!
